lib/vizkit/tree_view.rb

```diff
@@ -12,21 +12,35 @@ def self.setup_tree_view(tree_view)
         tree_view.connect(SIGNAL('customContextMenuRequested(const QPoint&)')) do |pos|
             index = tree_view.index_at(pos)
             next unless index.isValid
-            item = index.model.itemFromIndex index
+            if index.model.is_a?(Qt::AbstractProxyModel)
```

Comment:
You are copying this index-to-item code all over the place. Create a separate `item_from_index` method on the tree_view that factors out the functionality.